src/app/projects/components/project-settings/helper/data-handler-helper.ts

```diff
@@ -84,20 +87,40 @@ export class DataHandlerHelper {
         const values = settingModals.embedding.create.embeddingCreationFormGroup.getRawValue();
         let toReturn = this.getAttributeArrayAttribute(values.targetAttribute, 'name', attributes);
         toReturn += "-" + (values.granularity == 'ON_ATTRIBUTE' ? 'classification' : 'extraction');
-        toReturn += "-" + values.embeddingHandle;
-
+        const platform = settingModals.embedding.create.embeddingCreationFormGroup.get('platform').value
+        if (platform == PlatformType.HUGGING_FACE || platform == PlatformType.PYTHON) {
+            toReturn += "-" + platform + "-" + values.model;
+        } else if (platform == PlatformType.OPEN_AI) {
+            if (values.apiToken == null || values.apiToken.length == 0) {
+                toReturn += "-" + platform + "-" + values.model;
+            } else if (values.apiToken.length > 8) {
+                toReturn += "-" + values.model + "-" + values.apiToken?.substring(0, 8) + "...";
+            } else {
+                toReturn += "-" + platform + "-" + values.model + "-" + values.apiToken?.substring(0, values.apiToken.length);
+            }
+        } else if (platform == PlatformType.COHERE) {
+            if (values.apiToken == null || values.apiToken.length == 0) {
+                toReturn += "-" + platform + "-" + values.apiToken;
+            } else if (values.apiToken.length > 8) {
+                toReturn += "-" + platform + "-" + values.apiToken?.substring(0, 8) + "...";
```

Comment:
outdated logic for the api key.

Also this seems like a lot of similar code between the cases. 

Also I don't think we need to check for the length since substring of something with less length will result in the remaining string anyway.

Also also. why do we need to check `values.apiToken?` if we already know it exists (otherwise the length check before wouldn't work)